webapp/src/Controller/Jury/RejudgingController.php

```diff
@@ -685,4 +693,152 @@ private function generateFlashMessagesForSkippedJudgings(array $skipped): void
+            $runresults = [];
+            foreach ($curJudgings as $judging) {
+                if (!in_array($judging['result'], $results)) {
+                    $results[] = $judging['result'];
+                }
+                $judging_runs = $this->em->createQueryBuilder()
+                    ->from(JudgingRun::class, 'jr')
+                    ->select('t.rank', 'jr.runresult')
+                    ->leftJoin('jr.testcase', 't')
+                    ->andWhere('jr.judgingid = :judgingid')
+                    ->setParameter(':judgingid', $judging['judgingid'])
+                    ->orderBy('t.rank')
+                    ->getQuery()
+                    ->getArrayResult();
+                if (!in_array($judging_runs, $runresults)) {
+                    $runresults[] = $judging_runs;
+                }
+                $judging_results[$judging['judgingid']] = $judging['result'];
+            }
+            // If there are diffs on the judging level, then they will show up in the matrix anyway.
+            if (count($results) == 1) {
+                $submissions_to_result[$submitid] = $results[0];
+                if (count($runresults)!=1) {
+                    // Only report differences in judging runs if the final
+                    // results were the same.
+                    $judging_runs_differ[] = $submitid;
+                }
+            }
+
+            // Check for variations in runtimes across judgings
+            $runtimes = $this->em->createQueryBuilder()
+                ->from(JudgingRun::class, 'jr')
+                ->select('t.rank', 'MAX(jr.runtime) - MIN(jr.runtime) AS spread')
+                ->leftJoin('jr.judging', 'j')
+                ->leftJoin('jr.testcase', 't')
+                ->andWhere('j.submitid = :submitid')
+                ->setParameter(':submitid', $submitid)
+                ->groupBy('jr.testcaseid')
+                ->getQuery()
+                ->getArrayResult();
+            $runtime_spread[$submitid] = array(
+                'spread' => -1,
+                'rank' => -1
+            );
+            foreach ($runtimes as $runtime) {
+                $spread = (float) $runtime['spread'];
+                if ($spread > $runtime_spread[$submitid]['spread']) {
+                    $runtime_spread[$submitid]['spread'] = $spread;
+                    $runtime_spread[$submitid]['rank'] = $runtime['rank'];
+                    $runtime_spread[$submitid]['submitid'] = $submitid;
+                }
+            }
+        }
+        sort($judging_runs_differ);
+        usort($runtime_spread,
+            function ($a, $b) {
+                return $b['spread'] <=> $a['spread'];
+            }
+        );
+
+        $max_list_len = 10;
+        $runtime_spread_list = [];
+        $i = 0;
+        foreach ($runtime_spread as $value) {
+            if ($i >= $max_list_len) {
+                break;
+            }
+            $i++;
+            $submitid = $value['submitid'];
```

Comment:
This fails for me with an undefined index submitid. `$value` contains 
```php
     [
                'spread' => -1,
                'rank' => -1
     ]
```

Reply:
Fixed.